server/tests/test_history_persistence.py

```
@@ -0,0 +1,164 @@
+
+    monkeypatch.setattr("pathlib.Path.open", _boom)
+    h.add(_danmu("still-works"))  # 不應該拋出
+
+    assert len(h.get_records(limit=10)) == 1
+
+
+# ─── 端對端：真的走 HTTP /fire ────────────────────────────────────────────────
+
+
+def test_fire_endpoint_persists_to_disk(client, tmp_path, monkeypatch):
+    """POST /fire 之後，記錄要真的落在檔案裡。
```

Ruff (0.15.21)

[warning] Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)